src/inference/dev_api/cpp_interfaces/interface/ie_iplugin_internal.hpp

```diff
@@ -374,15 +372,169 @@ class INFERENCE_ENGINE_API_CLASS(IInferencePlugin) : public std::enable_shared_f
 
     std::string _pluginName;                            //!< A device name that plugins enables
     std::map<std::string, std::string> _config;         //!< A map config keys -> values
-    std::weak_ptr<ov::ICore> _core;                     //!< A pointer to ICore interface
+    std::weak_ptr<InferenceEngine::ICore> _core;        //!< A pointer to ICore interface
     std::shared_ptr<ExecutorManager> _executorManager;  //!< A tasks execution manager
     bool _isNewAPI;                                     //!< A flag which shows used API
 };
 
+/**
+ * @brief Class wraps InferenceEngine::IInferencePlugin into ov::IPlugin
+ */
+class INFERENCE_ENGINE_API_CLASS(IPluginWrapper) : public ov::IPlugin {
```

Comment:
Yes we need to have this class, because it should be exported from the openvino library.
RTTI won't work because convert_plugin is called from the plugin library runtime.